pallets/liquidity-pools-forwarder/src/lib.rs

```diff
@@ -0,0 +1,244 @@
+// Copyright 2024 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+//! # Liquidity Pools Forwarder Pallet.
+//!
+//! The Forwarder pallet acts as middleware for incoming and outgoing Liquidity
+//! Pools messages by wrapping them, if they are forwarded ones.
+//!
+//! For incoming messages, it extracts the payload from forwarded messages.
+//!
+//! For outgoing messages, it wraps the payload based on the configured router
+//! info.
+//!
+//! Assumptions: The EVM side ensures that incoming forwarded messages are
+//! valid.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+use cfg_traits::liquidity_pools::{LpMessage as LpMessageT, MessageReceiver, MessageSender};
+use cfg_types::domain_address::{Domain, DomainAddress};
+use frame_support::{dispatch::DispatchResult, pallet_prelude::*};
+use frame_system::pallet_prelude::OriginFor;
+pub use pallet::*;
+use parity_scale_codec::FullCodec;
+use sp_core::H160;
+use sp_std::convert::TryInto;
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct ForwardInfo {
+	/// Refers to the chain from which the message originates.
+	///
+	/// Example: Assume a three-hop A -> B -> C, then this refers to the domain
+	/// of A.
+	pub(crate) source_domain: Domain,
+	/// Refers to contract on forwarding chain.
+	///
+	/// Example: Assume A -> B -> C, then this refers to the forwarding
+	/// contract address on B.
+	pub(crate) contract: H160,
+}
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+	use std::fmt::Debug;
```

Comment:
Nit - we use `core::fmt:Debug` in other pallets.

Reply:
Fixed!